packages/plugins/documentation/server/controllers/documentation.js

```diff
@@ -98,6 +98,63 @@ module.exports = {
     }
   },
 
+  async json(ctx, next) {
+    try {
+      /**
+       * We don't expose the specs using koa-static or something else due to security reasons.
+       * That's why, we need to read the file localy and send the specs through it when we serve the Swagger UI.
+       */
+      const { major, minor, patch } = ctx.params;
+      const version =
+        major && minor && patch
+          ? `${major}.${minor}.${patch}`
+          : strapi.plugin('documentation').service('documentation').getDocumentationVersion();
+
+      const openAPISpecsPath = path.join(
+        strapi.dirs.app.extensions,
+        'documentation',
+        'documentation',
+        version,
+        'full_documentation.json'
+      );
+
+      try {
+        const documentation = fs.readFileSync(openAPISpecsPath, 'utf8');
+
+        try {
+          const layoutPath = path.resolve(
+            strapi.dirs.app.extensions,
+            'documentation',
+            'public',
+            'swagger.json'
+          );
+          await fs.ensureFile(layoutPath);
+          await fs.writeFile(layoutPath, documentation);
+
+          // Serve the file.
+          ctx.url = path.basename(`${ctx.url}/swagger.json`);
+
+          try {
+            const staticFolder = path.resolve(
+              strapi.dirs.app.extensions,
+              'documentation',
+              'public'
+            );
+            return koaStatic(staticFolder)(ctx, next);
+          } catch (e) {
+            strapi.log.error(e);
+          }
+        } catch (e) {
+          strapi.log.error(e);
+        }
+      } catch (e) {
+        strapi.log.error(e);
+      }
+    } catch (e) {
+      strapi.log.error(e);
+    }
```

Comment:
Hello !
That's a lot of `try...catch`. 🙂 
Could you reduce this to one, as it does exactly the same thing each time?

Also I think you can throw back an ApplicationError.

Reply:
Thanks for your response, I didn't really think we were having this try...catch in the other controller 😅 

If you don't mind, we would really appreciate it if you could extract common code into functions and rework this try...catch. 🙂

Reply:
@nathan-pichon I am not sure where I should place the [new function](https://github.com/strapi/strapi/pull/16698/files#diff-cf2afb5c6a87b10ab752d53d16c8b44f084cb7c701cb08457aa99c6cdabdd5a9R18-R39). 

Do you think the function should be in a new file, `server/services/utils/get-full-documentation.js` or is this placement fine? 
From my perspective, it doesn't make sense to place it next to services/utils functions, but I would like to know your advice.

Reply:
I think it's fine like that as it only concern controllers functions.

Reply:
So it's ready. ✅